### PR TITLE
Replace PyYAML with ruamel.yaml

### PR DESCRIPTION
**Pull request by abarto**
Replaced PyYAML with ruamel.yaml with round-trip set-up which allows us to respect the formatting and comments found on the user's config.yaml.

### CODE REVIEW

File: binstar_client/commands/config.py [hunk 1]
```diff
@@ -150,13 +150,12 @@ def main(args):
...
-                                      epilog=__doc__,
-                                      formatter_class=RawDescriptionHelpFormatter
-                                      )
+                                   help=description,
+                                   description=description,
+                                   epilog=__doc__,
+                                   formatter_class=RawDescriptionHelpFormatter)
 
-    parser.add_argument('--type', default=yaml.safe_load,
+    parser.add_argument('--type', default=str,
```
Comment (goanpeca) [inline binstar_client/commands/config.py]: Any Py2/Py3 problems that may arise from that `str`?

Comment (abarto) [reply binstar_client/commands/config.py]: Good point. I'll check it out.

Comment (abarto) [reply binstar_client/commands/config.py]: Fixed.

File: binstar_client/utils/yaml.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from .decorators import memoize
+
+
+@memoize
+def get_yaml():
+    """Tries to import the """
```
[the remark this answers is not in the document]
Comment (abarto) [reply binstar_client/utils/yaml.py]: Fixed.

File: record.txt [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+/home/abarto/miniconda3-anaconda-client/conda-bld/anaconda-client_1513711796965/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.6/site-packages/anaconda_client-1.6.6+5.gdec8a0b.dirty-py3.6.egg-info/PKG-INFO
+/home/abarto/miniconda3-anaconda-client/conda-bld/anaconda-client_1513711796965/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/bin/anaconda
+/home/abarto/miniconda3-anaconda-client/conda-bld/anaconda-client_1513711796965/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/bin/binstar
+/home/abarto/miniconda3-anaconda-client/conda-bld/anaconda-client_1513711796965/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/bin/conda-server
```
[the remark this answers is not in the document]
Comment (abarto) [reply record.txt]: Fixed.

File: conda.recipe/meta.yaml [hunk 4]
```diff
@@ -20,6 +20,7 @@ requirements:
     - python
     - setuptools
     - pytz
+    - future
```
Comment (goanpeca) [inline conda.recipe/meta.yaml]: Why are we mixing six and future ?

Seem like related libraries to accomplish something similar (albeit in a different way) Is this something we want to pursue?

Comment (abarto) [reply conda.recipe/meta.yaml]: Fixed

File: binstar_client/utils/yaml.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from .decorators import memoize
+
+
+@memoize
```
Comment (j0hn) [inline binstar_client/utils/yaml.py]: I think memoize is not really necessary here. We could remove the `get_yaml` function, move the import out and avoid calling the `get_yaml` on `yaml_load` and `yaml_dump` every time. Unless there's a reason to call the import every time i think that's the path i would follow.

Comment (abarto) [reply binstar_client/utils/yaml.py]: Fixed

**Pull request merged**